File: 0092.反转链表II/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseBetween(ListNode* head, int m, int n) {
        if (m==n)
            return head;
        ListNode * slow=new ListNode(0);
        slow->next=head;
        ListNode *fast=slow;
        for (int i=1;i<m;++i)
            fast=fast->next;
        ListNode *tail=fast->next,*p;
        for (int i=m;i<n;++i)
        {
            p=tail->next;
            tail->next=p->next;
            p->next=fast->next;
            fast->next=p;
        }
        return slow->next;
    }
};

File: 0092.反转链表II/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseBetween(head *ListNode, left int, right int) *ListNode {
	if left == right {
		return head
	}
	dummy := &ListNode{
		Next: head,
	}
	var st *ListNode = dummy
	var idx int
	for idx < left-1 {
		st = st.Next
		idx++
	}
	var ed *ListNode = st.Next
	for idx < right-1 {
		tar := ed.Next
		ed.Next = tar.Next
		tar.Next = st.Next
		st.Next = tar
		idx++
	}
	return dummy.Next
}
